### PR TITLE
new enrichment function: return test function following PU criteria

### PR DESCRIPTION
**Pull request by juliobarna**

### CODE REVIEW

File: kratos/utilities/enrichment_utilities.h [hunk 1]
```diff
@@ -1973,6 +1973,599 @@ class EnrichmentUtilities
     }
     
     
+	//2D: 2 enrichment functions for capturing weak and strong discontinities. All the shape functions follow the criteria of Partition of Unity.  
+	static int CalculateEnrichedShapeFuncionsExtendedModified2D(Geometry< Node<3> >& Julio,boost::numeric::ublas::bounded_matrix<double,(2+1), 2 >& rPoints, boost::numeric::ublas::bounded_matrix<double, (2+1), 2 >& DN_DX,array_1d<double,(2+1)>& rDistances, array_1d<double,(3*(2-1))>& rVolumes, boost::numeric::ublas::bounded_matrix<double, 3*(2-1), (2+1) >& rGPShapeFunctionValues, array_1d<double,(3*(2-1))>& rPartitionsSign, std::vector<Matrix>& rGradientsValue, boost::numeric::ublas::bounded_matrix<double,3*(2-1), (5)>& NEnriched,boost::numeric::ublas::bounded_matrix<double,10, 2>& rGradientpositive,boost::numeric::ublas::bounded_matrix<double,10, 2>& rGradientnegative ,boost::numeric::ublas::bounded_matrix<int,3,3>& father_nodes,std::vector<Matrix>& PRUEBA, array_1d<double,6>& weight)
```
Comment (pooyan-dadvand) [inline kratos/utilities/enrichment_utilities.h:1977]: I would suggest to change the parameter name Julio to some more descriptive name

**Comment by loumalouomega** [comment]
@juliobarna Like this is a change concerning the core I added to @pooyan-dadvand  and @RiccardoRossi for review the code

File: kratos/utilities/enrichment_utilities.h [hunk 2]
```diff
@@ -1973,6 +1973,599 @@ class EnrichmentUtilities
...
+	      //type_of_cut=1;
+	      for (int j = 0; j < 2; j++)
+                rGradientsValue[0](0, j) = 0.0;
+	      if (rDistances(0) < 0.0) rPartitionsSign[0] = -1.0;
+	      else rPartitionsSign[0] = 1.0;
+	      //KRATOS_WATCH("one element not in the intefase")
+			return 1;
+	    }
+
+	  //const double unsigned_distance0=fabs(rDistances(0));
```
Comment (pooyan-dadvand) [inline kratos/utilities/enrichment_utilities.h:2011]: I would remove the commented lines. They will be in your history.

**Review by pooyan-dadvand — CHANGES_REQUESTED** [review]
Apart from my comments in the code, do you have any test for this?
Meanwhile I see a lot of copy paste in this method and in the file in general. I think @RiccardoRossi can give you some hint about avoiding such a copy.

File: kratos/utilities/enrichment_utilities.h (hunk 1 above)
Comment (juliobarna) [reply kratos/utilities/enrichment_utilities.h:1977]: Thanks for the comments. This function could be used to enrich temperature or velocity. For this reason, an example can be incorporated in the convection-diffusion application.

Comment (loumalouomega) [reply kratos/utilities/enrichment_utilities.h:1977]: @juliobarna You can add tests additionally to the examples, this way the correct functioning of the utility can be checked nightly, see https://github.com/KratosMultiphysics/Kratos/wiki/How-to-create-unitary-tests

**Pull request closed without merging**